### PR TITLE
ports: add documentation page

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
Add a documentation page devoted to the concept of ports in Tarantool.

The goal of this document is to enhance the understanding of ports and their role in facilitating dataflow within Tarantool.

The document covers the following key areas:
* introduction to ports;
* general overview of ports;
* ports interface;
* dataflow with ports.

### CODE REVIEW

**Review by Gumix — APPROVED** [review]

File: source/ports.rst [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+The MsgPack port is created for processing IPROTO call (eval) requests
+(i.e., RPC). The arguments for procedure call arguments are prepared using this
+port type. It is the simplest port of all, plainly storing MsgPack.
+
+The VDBE port is used for calling stored procedures from inside the VDBE. It is
+stores an array of VDBE memory registers.
+
+Last but not least, the SQL port is needed for storing the results of SQL
+queries and SQL expressions. It stores C port data together with the original
+SQL statement needed for retrieving SQL metadata later on.
```
Comment (Totktonada) [inline source/ports.rst:154]: I would list source files, where the given ports are implemented or a name of the structure. Something that would simplify matching these descriptions with the code.

Comment (CuriousGeorgiy) [reply source/ports.rst:154]: Supplied all ports with structure names, thanks for this reasonable suggestion.

Comment (Totktonada) [reply source/ports.rst:154]: Forgot to push?

Comment (CuriousGeorgiy) [reply source/ports.rst]: True.

**Review by alyapunov — APPROVED** [review]

**Pull request merged**